src/v/utils/tests/external_process_test.cc

```diff
@@ -0,0 +1,202 @@
+                      {"/bin/echo", "-n", "hi"})
+                      .get();
+
+        proc.set_stdout_consumer(consumer{"hi", matched});
+        auto res = proc.wait().get();
+
+        BOOST_REQUIRE(
+          std::holds_alternative<ss::experimental::process::wait_exited>(res));
+        auto wait_exited = std::get<ss::experimental::process::wait_exited>(
+          res);
+        BOOST_CHECK_EQUAL(wait_exited.exit_code, 0);
+
+        BOOST_CHECK(matched);
+        return ss::now();
+    }).get();
+}
+
+SEASTAR_THREAD_TEST_CASE(external_process_run_long_process) {
+    using namespace std::chrono_literals;
+
+    auto proc = external_process::external_process::create_external_process(
+                  {"/bin/sleep", "10"})
+                  .get();
+    auto start_time = std::chrono::high_resolution_clock::now();
+    auto res = proc.wait().get();
+    auto end_time = std::chrono::high_resolution_clock::now();
+    BOOST_REQUIRE(
+      std::holds_alternative<ss::experimental::process::wait_exited>(res));
+    auto wait_exited = std::get<ss::experimental::process::wait_exited>(res);
+    BOOST_CHECK_EQUAL(wait_exited.exit_code, 0);
+    BOOST_CHECK(
+      std::chrono::duration_cast<std::chrono::seconds>(end_time - start_time)
+        .count()
+      >= 0);
+}
+
+SEASTAR_THREAD_TEST_CASE(external_process_run_long_process_and_terminate) {
+    using namespace std::chrono_literals;
+
+    auto proc = external_process::external_process::create_external_process(
+                  {"/bin/sleep", "10"})
+                  .get();
+    auto wait_fut = proc.wait();
+    ss::sleep(1s).get();
+    BOOST_CHECK(proc.is_running());
+    proc.terminate(1s).get();
+    BOOST_CHECK(!proc.is_running());
+    auto res = wait_fut.get();
+    BOOST_REQUIRE(
+      std::holds_alternative<ss::experimental::process::wait_signaled>(res));
+    auto wait_signaled = std::get<ss::experimental::process::wait_signaled>(
+      res);
+    BOOST_CHECK_EQUAL(wait_signaled.terminating_signal, SIGTERM);
+}
+
+SEASTAR_THREAD_TEST_CASE(external_process_test_sigterm_ignored) {
+    using namespace std::chrono_literals;
+    const char* script_path = std::getenv("HANDLE_SIGTERM_SCRIPT");
+    vassert(script_path, "Missing 'HANDLE_SIGTERM_SCRIPT' env variable");
+
+    BOOST_REQUIRE_MESSAGE(
+      ss::file_exists(script_path).get(),
+      std::string(script_path) + " does not exist");
+
+    ss::do_with(
+      bool(false),
+      std::string(script_path),
+      [](auto& matched, auto& script_path) {
+          auto proc
+            = external_process::external_process::create_external_process(
+                {script_path.c_str()})
+                .get();
+          proc.set_stdout_consumer(consumer{"sigterm called", matched});
+          ss::sleep(100ms).get();
+          BOOST_REQUIRE(proc.is_running());
+          auto wait_fut = proc.wait();
+          auto term_fut = proc.terminate(5s);
+          ss::sleep(1s).get();
```

Comment:
While this seems fine, have you ran this test 1000 times to make sure it doesn't flake?

Reply:
I could do it 10,000 times if you like